s3tests_boto3/functional/test_s3.py

```diff
@@ -169,6 +208,24 @@ def test_bucket_list_delimiter_basic():
     eq(len(prefixes), 2)
     eq(prefixes, ['foo/', 'quux/'])
 
+@attr(resource='bucket')
+@attr(method='get')
+@attr(operation='list')
+@attr(assertion='prefixes in multi-component object names')
+@attr('list-objects-v2')
+def test_bucket_listv2_delimiter_basic():
+    bucket_name = _create_objects(keys=['foo/bar', 'foo/bar/xyzzy', 'quux/thud', 'asdf'])
+    client = get_client()
+
+    response = client.list_objects_v2(Bucket=bucket_name, Delimiter='/')
+    eq(response['Delimiter'], '/')
+    keys = _get_keys(response)
+    eq(keys, ['asdf'])
+
+    prefixes = _get_prefixes(response)
+    eq(len(prefixes), 2)
+    eq(prefixes, ['foo/', 'quux/'])
+
 def validate_bucket_list(bucket_name, prefix, delimiter, marker, max_keys,
```

Comment:
@albIN7 there needs to be made another version of this function for list_object_v2. This function is called in `test_bucket_list_delimiter_prefix, test_bucket_list_delimiter_prefix_ends_with_delimiter, test_bucket_list_delimiter_prefix_underscore` all which need v2 versions.

Reply:
Added